src/fifty_points_forex/main.py

```diff
@@ -0,0 +1,178 @@
+)
+from immediate_execution_model import ImmediateExecutionModel
+from pytz import timezone
+from day_of_week import DayOfWeek
+
+# endregion
+
+
+class OcODevisenStrategy(QCAlgorithm):
+
+    def initialize(self):
+        self.set_start_date(2024, 1, 1)
+        self.set_end_date(2024, 11, 11)
+
+        self.set_cash(100000)
+        self.default_order_properties.time_in_force = TimeInForce.DAY
+        time_zone_id = "Europe/Berlin"
+        self.berlin_tzinfo = timezone(time_zone_id)
+        self.set_time_zone(time_zone_id)
+
+        self.set_brokerage_model(
+            brokerage=BrokerageName.INTERACTIVE_BROKERS_BROKERAGE,
+            account_type=AccountType.MARGIN,
+        )
+
+        symbols: List[Symbol] = [
+            self.add_forex(ticker=currency_pair, resolution=Resolution.MINUTE).symbol
+            for currency_pair in ["EURUSD", "GBPUSD", "EURGBP"]
+            # for currency_pair in ["EURUSD"]
+        ]
+
+        self.add_universe_selection(ManualUniverseSelectionModel(symbols))
+        self.universe_settings.resolution = Resolution.MINUTE
+
+        self.symbol_data: Mapping[Symbol, SymbolData] = {}
+
+        self.pip = 0.0001  # TODO make this dependend on currency pair, this is not correct for Yen
+        self.lot_size = 100000
+        self.orders = {}
+        # self.add_risk_management(TrailingStopRiskManagementModel(0.006))
+        self.risk_management_model = OneRTrailingStopRiskManagementModel()
+        self.add_risk_management(self.risk_management_model)
+        self.set_execution(ImmediateExecutionModel())
+
+    def on_hourly_quote_bar(self, sender, quote_bar: QuoteBar):
+        self.symbol_data[quote_bar.symbol].last_hour_quote_bar = quote_bar
+
+    def get_quantity(self, quote_bar: QuoteBar) -> Mapping[str, float]:
+        def get_maximum_loss(price1: float, price2: float) -> float:
+            maximum_loss = round(price1 - price2, 6)
+            return maximum_loss if maximum_loss != 0 else 6 * self.pip
+
+        def calculate_max_margin() -> float:
+            available_margin = self.portfolio.margin_remaining
+            invested_count = sum(
+                1 if self.portfolio[symbol].invested else 0
+                for symbol in self.symbol_data
+            )
+
+            # Calculate margin ratio based on current investments
+            margin_ratio = max(3 - invested_count, 1)
+            return available_margin / margin_ratio
+
+        def calculate_position_size(
+            max_margin: float, risk_exposure: float, max_loss: float
+        ) -> float:
+            position_size = risk_exposure / max_loss
+            desired_size = min(position_size, max_margin)
+            return max(round(desired_size / self.lot_size), 1) * self.lot_size
+
+        risk_exposure = self.portfolio.total_portfolio_value * 0.01
+
+        # Calculate maximum potential loss for buy and sell
+        maximum_loss_buy = get_maximum_loss(quote_bar.close, quote_bar.low)
```

Comment:
this is not correct , we need to take the actual stop loss distance which is high - low